File: src/java/src/quickfix/fix50/Message.java
package quickfix.fix50;
import quickfix.FieldNotFound;
import quickfix.Group;
import quickfix.field.*;

public class Message extends quickfix.Message {
  public Message() {
    super();
    header = new Header( this );
    trailer = new Trailer( this );
    getHeader().setField(new BeginString("FIX.5.0"));
  }
  public class Header extends quickfix.Message.Header {
    public Header( Message message ) {
      super( message );
    }
  }

}

File: src/.NET/fix50sp1/Message.cs
namespace QuickFix50Sp1
{
  public class Header : QuickFix.Message.Header
  {
    Header(QuickFix.Message message) : base(message) {}

  };

  public class Trailer : QuickFix.Message.Trailer
  {
    Trailer(QuickFix.Message message) : base(message) {}

  };

  public class Message : QuickFix.Message
  {
    public Message() : base(new QuickFix.BeginString("FIXT.1.1"))
    {
      m_header = new Header(this);
      m_trailer = new Trailer(this);
    }

    public Message( QuickFix.MsgType msgType ) : base(new QuickFix.BeginString("FIXT.1.1"), msgType)
    {
      m_header = new Header(this);
      m_trailer = new Trailer(this);
    }

    public new Header getHeader() { return (Header)(base.getHeader()); }
    public new Trailer getTrailer() { return (Trailer)(base.getTrailer()); }
  };

}

